daemon/Jobs/UpdaterJob.cs

```diff
@@ -135,29 +195,60 @@ public void Perform()
             AppConfig.UpdateDeadlock = true;
 
             // Get the latest set of release data.
-            var releaseInformation = GetReleaseInformation();
+            releaseInformation = GetReleaseInformation();
+            
+            // Determine the release channel 
+            runningBranch = _config.Updater.ReleaseChannel ?? AppConfig.ReleaseChannel;
+            _logger.LogDebug("The running branch is " + runningBranch);
+
+            // Validate that the release channel exists.
+            if (!_validReleaseChannels.Contains(runningBranch))
+            {
+                // Let the user know
+                _logger.LogWarning("UJ: The release channel {0} does not support updates - updating will be disabled.",
+                    runningBranch);
 
-            // Get Version details.
-            var runningBranch = _config.Updater.ReleaseChannel ?? AppConfig.ReleaseChannel;
-            var remoteVersion = releaseInformation.channels[runningBranch];
-            var remoteBranch = remoteVersion.Split("-")[1];
+                // Disable updating
+                _config.Updater.Enabled = false;
 
+                // Stop execution
+                return;
+            }
 
-            // Compare
+            // Get the remainder versioning information
+            remoteVersion = releaseInformation.channels[runningBranch];
+            remoteBranch = remoteVersion.Split("-")[1];
+
+            // Compare and make sure there's an update available.
             if (remoteBranch == runningBranch && SemanticVersionUpdateChecker(remoteVersion))
             {
                 // Update available.
-                var newVersion = releaseInformation.channels[remoteBranch];
-                var targetDirectory = Path.Combine(RootInstallationDirectory, newVersion);
-                var targetArchive = Path.Combine(RootInstallationDirectory, string.Format("{0}.zip", newVersion));
+                _logger.LogDebug("UJ: Generating target paths...");
```

Comment:
I like the step by step logging, keep it up.

Reply:
This was rubber ducking, But alright.